diff -up mesa-25.0.7/src/gallium/auxiliary/gallivm/lp_bld_nir_soa.c.dma mesa-25.0.7/src/gallium/auxiliary/gallivm/lp_bld_nir_soa.c
--- mesa-25.0.7/src/gallium/auxiliary/gallivm/lp_bld_nir_soa.c.dma	2025-05-29 01:20:23.000000000 +1000
+++ mesa-25.0.7/src/gallium/auxiliary/gallivm/lp_bld_nir_soa.c	2025-09-26 12:32:52.240568901 +1000
@@ -1321,8 +1321,10 @@ emit_load_const(struct lp_build_nir_cont
 
    for (unsigned i = 0; i < instr->def.num_components; i++) {
      outval[i] = lp_build_const_int_vec(bld_base->base.gallivm, int_bld->type,
-                                        bits == 32 ? instr->value[i].u32
-                                                   : instr->value[i].u64);
+                                        bits == 8 ? instr->value[i].u8 :
+                                        bits == 16 ? instr->value[i].u16 :
+                                        bits == 32 ? instr->value[i].u32 :
+                                                     instr->value[i].u64);
    }
    for (unsigned i = instr->def.num_components; i < NIR_MAX_VEC_COMPONENTS; i++) {
       outval[i] = NULL;
